src/main.cpp

```diff
@@ -165,73 +173,46 @@ int load_conf_file(char *file_path)
     return 0;
 }
 
-static void run_test(int test_n) {
-    for (int i = 0; i < tests_exclude.size; i++) {
-        // If a match is found in the exclusion list, then we skip the test.
-        if (test_n == vector_get(&tests_exclude, i)) {
-            test_n = -1;
-            break;
-        }
-    }
-    // Skip the test if test_n is a negative number.
-    if (test_n >= 0) {
-        kernel_api_tests[test_n].func(test_n, kernel_api_tests[test_n].name);
-    }
-}
-
 static void run_tests()
 {
     print("Random seed used is %u", seed);
-    if (tests_to_run.size == 0) {
-        print("No specific tests were requested. Running all tests (Single Pass).");
-        if (tests_exclude.size) {
-            int exclude_count = 0;
-            for (int i = 0; i < tests_exclude.size; i++) {
-                if (kernel_api_tests_size > vector_get(&tests_exclude, i)) {
-                    exclude_count++;
-                }
-            }
-            print("%d test(s) will be excluded.", exclude_count);
-        }
-        print("-------------------------------------------------------------");
-        for (int k = 0; k < kernel_api_tests_size; k++) {
-            run_test(k);
+    if (tests_to_run.none()) {
+        print("No specific tests were requested. Running all tests.");
+        // Enable all tests.
+        tests_to_run.flip();
+        if (tests_to_exclude.any()) {
+            // Exclude any tests requested.
+            tests_to_run &= ~tests_to_exclude;
+            print("%zu test(s) will be excluded.", tests_to_exclude.count());
         }
     }
     else {
         print("A config file was loaded. Only running requested tests.");
-        if (tests_exclude.size) {
-            int exclude_count = 0;
-            for (int k = 0; k < tests_to_run.size; k++) {
-                int test_n = vector_get(&tests_to_run, k);
-                for (int i = 0; i < tests_exclude.size; i++) {
-                    if (test_n == vector_get(&tests_exclude, i)) {
-                        exclude_count++;
-                        break;
-                    }
-                }
-            }
-            print("%d test(s) will be excluded.", exclude_count);
+        if (tests_to_exclude.any()) {
+            const auto tests_to_run_excluded = tests_to_run & tests_to_exclude;
+            // Exclude any tests requested.
+            tests_to_run &= ~tests_to_run_excluded;
+            print("%zu test(s) will be excluded.", tests_to_run_excluded.count());
         }
-        print("-----------------------------------------------------");
-        for (int k = 0; k < tests_to_run.size; k++) {
-            run_test(vector_get(&tests_to_run, k));
+    }
+    print("-------------------------------------------------------------");
+    for (int i = 0; i < kernel_api_tests_size; i++) {
+        if (tests_to_run.test(i)) {
```

Comment:
For unknown reason, my previous remark got misplaced;
What I intended was to ask if `if (tests_to_run.test(i))` could be slightly changed into `if (tests_to_run[i])`

Reply:
According to the bitset's doc:
Test caller can throw out of range exception. Whilst [] operator has undefined behavior for out of bounds. In my honest opinion, using the test function provided by bitset is better to use in case of need to fix up either here (unlikely) or in the future codes using bitset to improve our skillset for using the standard bitset library.

bitset refs:
- [test(position/index)](https://en.cppreference.com/w/cpp/utility/bitset/test)
- [operator[]](https://en.cppreference.com/w/cpp/utility/bitset/operator_at)

Reply:
Fine by me.